source/val/validate_extensions.cpp

```diff
@@ -2028,6 +2131,377 @@ spv_result_t ValidateExtInst(ValidationState_t& _, const Instruction* inst) {
+        auto validate_parent =
+            ValidateOperandLexicalScope(_, "Parent", inst, 10, ext_inst_name);
+        if (validate_parent != SPV_SUCCESS) return validate_parent;
+        break;
+      }
+      case OpenCLDebugInfo100DebugTypeFunction: {
+        auto* return_type = _.FindDef(inst->word(6));
+        if (return_type->opcode() != SpvOpTypeVoid) {
+          auto validate_return = ValidateOperandDebugType(
+              _, "Return Type", inst, 6, ext_inst_name);
+          if (validate_return != SPV_SUCCESS) return validate_return;
+        }
+        for (uint32_t word_index = 7; word_index <= num_operands;
+             ++word_index) {
+          auto validate_param = ValidateOperandDebugType(
+              _, "Parameter Types", inst, word_index, ext_inst_name);
+          if (validate_param != SPV_SUCCESS) return validate_param;
+        }
+        break;
+      }
+      case OpenCLDebugInfo100DebugTypeEnum: {
+        auto validate_name = ValidateOperandForDebugInfo(
+            _, "Name", SpvOpString, inst, 5, ext_inst_name);
+        if (validate_name != SPV_SUCCESS) return validate_name;
+        auto validate_underlying_type = ValidateDebugInfoOperand(
+            _,
+            [](OpenCLDebugInfo100Instructions dbg_inst) {
+              return dbg_inst == OpenCLDebugInfo100DebugInfoNone;
+            },
+            inst, 6);
+        if (validate_underlying_type != SPV_SUCCESS) {
+          validate_underlying_type = ValidateOperandDebugType(
+              _, "Underlying Types", inst, 6, ext_inst_name);
+          if (validate_underlying_type != SPV_SUCCESS)
+            return validate_underlying_type;
+        }
+        auto validate_src = ValidateDebugInfoOperand(
+            _, "Source", OpenCLDebugInfo100DebugSource, inst, 7, ext_inst_name);
+        if (validate_src != SPV_SUCCESS) return validate_src;
+        auto validate_parent =
+            ValidateOperandLexicalScope(_, "Parent", inst, 10, ext_inst_name);
+        if (validate_parent != SPV_SUCCESS) return validate_parent;
+        auto validate_size = ValidateOperandForDebugInfo(
+            _, "Size", SpvOpConstant, inst, 11, ext_inst_name);
+        if (validate_size != SPV_SUCCESS) return validate_size;
+        for (uint32_t word_index = 13; word_index <= num_operands;
+             word_index += 2) {
+          auto validate_value = ValidateOperandForDebugInfo(
+              _, "Value", SpvOpConstant, inst, word_index, ext_inst_name);
+          if (validate_value != SPV_SUCCESS) return validate_value;
+          assert(word_index + 1 <= num_operands && "Missing \"Name\" operand");
+          validate_name = ValidateOperandForDebugInfo(
+              _, "Name", SpvOpString, inst, word_index + 1, ext_inst_name);
+          if (validate_name != SPV_SUCCESS) return validate_name;
+        }
+        break;
+      }
+      case OpenCLDebugInfo100DebugTypeComposite: {
+        auto validate_name = ValidateOperandForDebugInfo(
```

Comment:
This is a very repetitive code sequence.  The same things are repeatedly done:
  - call a helper method to check
  - check return status, and return the status if it's not SPV_SUCCESS
  - in these things, reuse inst and ext_inst_name.

I think you can shorten and clarify by using macros, e.g. near the top, define:
   #define CHECK_OPERAND(NAME, opcode, index)  { \
      auto result = ValidateperandForDebugInfo(_, NAME, opcode, inst, index, ext_inst_name); if (result != SPV_SUCCESS) return result;}

Then at this point you would only write this:
  CHECK_OPERAND("Name", SpvOpString, 5)


Please shorten all your code sequences. It will make the code far more readable and easier to verify.

Reply:
done